plugins/mysql/mysql_audit

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+# -*- python -*-
+
+"""
+=head1 INTRODUCTION
+
+Plugin to monitor the MySQL audit log connection count
+
+=head1 APPLICABLE SYSTEMS
+
+MySQL needs to be configured manually in order to enable the audit log. This
+can be done as follows:
+
+=over 2
+
+      plugin-load = server_audit=server_audit.so
+      server_audit_events = connect
+      server_audit_output_type=file
+      server_audit_file_path = /var/log/mysql/audit.log
+      server_audit_file_rotate_size = 512M
+      server_audit_logging = ON
+      server_audit_file_rotations = 5
+
+=back
+
+=head1 INSTALLATION
+
+Place in /etc/munin/plugins/ (or link it there using ln -s)
+
+=head1 CONFIGURATION
+
+Add this to your /etc/munin/plugin-conf.d/munin-node:
+
+=over 2
+
+      [mysql_audit]
+      user mysql
+      env.logfile /var/log/mysql/audit.log
+      env.sorted 1 # sort output (if not: unsorted)
+      env.toplist 10 # only the top 10 (if unset: all of them). implies sorted
+
+=back
+
+=head1 HISTORY
+
+2017-11-03: v 1.0 Bert Van de Poel <bert@bhack.net>: created
+2020-07-19: v 1.1 pcy <pcy@ulyssis.org>: added config options
+
+=head1 USAGE
+
+Parameters understood:
+
+  config   (required)
+  autoconf (optional - used by munin-config)
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf
+"""
+
+
+from datetime import datetime, timedelta, timezone
+import operator
+import os
+import sys
+
+
+def weakbool(x):
+    return x.lower().strip() in {'true', 'yes', 'y', '1'}
+
+
+logfile = os.getenv('logfile', '/var/log/mysql/audit.log')
+toplist = int(os.getenv('toplist', '0'))
+sortlist = weakbool(os.getenv('sorted', 'N')) or toplist > 0
+
+STATEFILE = os.getenv('MUNIN_STATEFILE')
+STATE = None
+
+
+if os.path.isfile(STATEFILE):
```

Comment:
I think, it would be good to move this to a `load_state` function and only call it, when it is necessary (avoiding `global`).

Additionally it would probably be much simpler to store the file position instead of a timestamp. But this is a matter of taste.

Reply:
Moved it to a function.

As I argued [above](https://github.com/munin-monitoring/contrib/pull/1104#discussion_r471213477), I don't think it won't cause unnecessary parsing etc., and because it works right now, I don't want to mess with it much.